aptos/contract/aux/sources/reward_quoter.move

```diff
@@ -0,0 +1,349 @@
+// Auto generated from "go-util/aptos/cmd/gen-reward-distributor"
```

Comment:
What kind of calculations?

Reply:
> What kind of calculations?

For example, in stable swap, user can get a quoter from stable swap and calculate how much fees they can get from their lp holdings.

Reply:
Could we just implement functions for that on the `RewardDistributor`?

Reply:
> Could we just implement functions for that on the `RewardDistributor`?

Not really. `RewardDistributor` contains `Coin`, so cannot be copied/passed around easily. A separate `struct` is almost unavoidable. We can replicate the logic with integer operations, but it will be easier just to use `quote_coin` and simply duplicate the code.

Reply:
Ok, maybe I don't fully understand the use case. It seems like just implementing a function to do the math with integers would make more sense.

Reply:
> Ok, maybe I don't fully understand the use case. It seems like just implementing a function to do the math with integers would make more sense.

That will mean similar but not exactly the same operations, and we need to maintain the two version in sync. By auto generating the code, those two versions will always be exactly the same logically.

Reply:
I see, but it seems like this introduces more operational complexity/overhead. It seems like we probably won't be changing the mathematical operations much, so keeping them in sync wouldn't be too difficult. My preference would be to keep the move code as minimal as possible in general, and only use code gen if it's totally necessary so it's more straightforward how to interact with the codebase.

Reply:
> But I don't totally understand what the calculations we would expect people to do are... like they just are simulating what would happen if a certain amount of fees were deposited and they had a certain stake? seems pretty complicated to simulate anyway because how would they know what the fees are?

Some examples are here [quoter_3pool](https://github.com/aux-exchange/aux-exchange/blob/stable-pool/aptos/contract/aux/sources/quoter_3pool_test.move). The basic idea is that we simply replaces `RewardDistributor` with `RewardDistributionQuoter` and nothing else needs to change.

Another example is the implementation of `au_in` through this interface [here](https://github.com/aux-exchange/aux-exchange/blob/stable-pool/aptos/contract/aux/sources/stable_2pool_test.move#L54-L66).